### PR TITLE
Add Cognito authorization vi Amplify sdk and related compensating changes.

### PR DESCRIPTION
**Pull request by eriksendc**
Summary
---------

Created an AWS Cognito user pool, and there is now authentication to the web app and the api calls the web app makes.

Test Plan
---------

1. terraform destroy
2. terraform apply
3. npm start
4. exercise /matches and /upload

### CODE REVIEW

File: hasher-matcher-actioner/terraform/api/main.tf [hunk 1]
```diff
@@ -104,77 +104,6 @@ resource "aws_iam_role_policy_attachment" "api_root" {
   policy_arn = aws_iam_policy.api_root.arn
 }
 
-
-# Authorizer API Lambda
-
```
Comment (eriksendc) [inline hasher-matcher-actioner/terraform/api/main.tf:109]: Note that I didn't delete the python source for the authorizer... sort of on purpose. was thinking to ask for comments on that.

File: hasher-matcher-actioner/terraform/api/main.tf [hunk 2]
```diff
@@ -223,14 +160,14 @@ resource "aws_apigatewayv2_integration" "hma_apigateway" {
...
-  authorizer_payload_format_version = "2.0"
-  enable_simple_responses = true
-  authorizer_result_ttl_in_seconds = 0
-  name             = "${aws_apigatewayv2_api.hma_apigateway.name}_authorizer"
+  authorizer_type  = "JWT"
+  identity_sources = ["$request.header.Authorization"]
+  name             = "${var.prefix}-jwt-authorizer"
+
+  jwt_configuration {
+    audience = ["1hu072n2n920b7kadllfi3imko"]
```
Comment (BarrettOlson) [inline hasher-matcher-actioner/terraform/api/main.tf:168]: did you mean to pass the variables all the way down to here?

File: hasher-matcher-actioner/terraform/api/main.tf [hunk 3]
```diff
@@ -104,77 +104,6 @@ resource "aws_iam_role_policy_attachment" "api_root" {
   policy_arn = aws_iam_policy.api_root.arn
 }
 
-
-# Authorizer API Lambda
```
Comment (BarrettOlson) [inline hasher-matcher-actioner/terraform/api/main.tf:108]: nice! can you delete the python code too?

File: hasher-matcher-actioner/terraform/api/main.tf (hunk 2 above)
Comment (eriksendc) [reply hasher-matcher-actioner/terraform/api/main.tf:168]: Doh! Yes I set up all those nice variables and then didn't use them!!!!

**Review by BarrettOlson — APPROVED** [review]
Accepting to unblock. 

Please either update contributing and/or webapp/README with the steps needed to be able to run your test plan or add a Make command to do it ~automatically before merge (the .ENV stuff). 

And like @Dcallies said the sooner we can make even these cognito pieces ~easily deployable the better. :) Thanks again for working on this!

File: hasher-matcher-actioner/terraform/api/main.tf (hunk 2 above)
Comment (eriksendc) [reply hasher-matcher-actioner/terraform/api/main.tf]: Fixed!

File: hasher-matcher-actioner/terraform/api/main.tf (hunk 1 above)
Comment (eriksendc) [reply hasher-matcher-actioner/terraform/api/main.tf:109]: We decided to delete api_auth.py.

File: hasher-matcher-actioner/terraform/api/main.tf (hunk 3 above)
Comment (eriksendc) [reply hasher-matcher-actioner/terraform/api/main.tf:108]: absolutely!

File: hasher-matcher-actioner/webapp/README.md [hunk 4]
```diff
@@ -2,7 +2,16 @@
 
 # Getting started with the hasher-matcher-actioner webapp
 
-This project requires [npm](https://www.npmjs.com/get-npm) be installed. Once installed  `npm install` from within this directory should enable usage of the scripts referenced below.
+This project requires [npm](https://www.npmjs.com/get-npm) be installed. Once installed `npm install` from within this directory should enable usage of the scripts referenced below.
+
+To run, build or test you need to create a .env file in the webapp directory. One is created for you when you terraform apply, and it should look something like what you see in .env.example.
+
+To sign in (whether running from localhost, the s3 bucket url, or the cloudfront distribution) you need a user. Take these steps to create a user after you've run terraform apply:
```
Comment (BarrettOlson) [inline hasher-matcher-actioner/webapp/README.md:9]: ❤️, think is it worth updating CONTRIBUTING.md with something like "For the UI/Webapp components see `webapp/README` ?

File: hasher-matcher-actioner/terraform/authentication/main.tf [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+resource "aws_cognito_user_pool_client" "webapp_and_api_user_pool_client" {
+  name                                 = "${var.prefix}-user-pool-client"
+  user_pool_id                         = aws_cognito_user_pool.webapp_and_api_user_pool.id
+  generate_secret                      = false
+  allowed_oauth_flows_user_pool_client = true
+  explicit_auth_flows                  = ["ALLOW_CUSTOM_AUTH", "ALLOW_USER_SRP_AUTH", "ALLOW_REFRESH_TOKEN_AUTH"]
+  allowed_oauth_scopes                 = ["openid"]
+  allowed_oauth_flows                  = ["code"]
+  callback_urls                        = ["https://localhost:3000"]
```
Comment (BarrettOlson) [inline hasher-matcher-actioner/terraform/authentication/main.tf:50]: Just curious, how do these work? can we support them both locally and using cloudfront? Thanks!

**Pull request merged**